### PR TITLE
Add initial support for Silicon Labs EFM32PG-STK3402A board

### PR DESCRIPTION
**Pull request by chrta**
The following examples are tested:

- hello_world
- basic/blinky
- basic/button
- subsys/shell/shell_module

This continues #7043, which can be closed now.

### CODE REVIEW

File: boards/arm/efm32pg_stk3402a/board.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2018 Christian Taedcke
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __INC_BOARD_H
+#define __INC_BOARD_H
+
+#include <soc.h>
```
Comment (galak) [inline boards/arm/efm32pg_stk3402a/board.h]: can we remove <soc.h>

File: boards/arm/efm32pg_stk3402a/board.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2018 Christian Taedcke
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <board.h>
```
Comment (galak) [inline boards/arm/efm32pg_stk3402a/board.c]: "board.h"

File: boards/arm/efm32pg_stk3402a/board.h (hunk 1 above)
Comment (chrta) [reply boards/arm/efm32pg_stk3402a/board.h]: done

File: boards/arm/efm32pg_stk3402a/board.c (hunk 2 above)
Comment (chrta) [reply boards/arm/efm32pg_stk3402a/board.c]: done

**Review by dbkinder — APPROVED** [review]
+1 for doc addition, thanks.

**Comment by chrta** [comment]
Added flash support and rebased on current master.

**Review by galak — CHANGES_REQUESTED** [review]
Need to rework dts_fixup.h for recent CONFIG_ -> DT_ change

File: boards/arm/efm32pg_stk3402a/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+config CMU_HFXO_FREQ
+	default 40000000
+
+config CMU_LFXO_FREQ
+	default 32768
+
+if GPIO_GECKO
+
+config GPIO_GECKO_PORTA
+	def_bool y
```
Comment (galak) [inline boards/arm/efm32pg_stk3402a/Kconfig.defconfig]: replace def_bool y with 'default y'

Comment (chrta) [reply boards/arm/efm32pg_stk3402a/Kconfig.defconfig]: done

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
Can you fixup the missing copyright/license on:

dts/arm/silabs/efm32pg12b.dtsi missing license.
dts/arm/silabs/efm32pg12b.dtsi missing copyright.

I'll chat with @nashif about the zlib license checks.

**Comment by chrta** [comment]
@galak Ok, i will add this license and copyright info to this file.

**Comment by chrta** [comment]
@galak @nashif The same licenses are also used in the newest gecko sdk in #11453

**Review by galak — APPROVED** [review]

**Pull request merged**